src/rand.jl

```diff
@@ -0,0 +1,42 @@
+import Random
+
+# TODO: remove Random.CloseOpen01{BigFloat} parameters
+# TODO: remove setprecision
+# possibly fixed by https://github.com/JuliaLang/julia/pull/38169
+
+function Random.Sampler(
+    RNG::Type{<:Random.AbstractRNG},
+    st::Random.SamplerType{T},
+    n::Random.Repetition,
+) where {T<:Union{Arf,Arb,Acb}}
+    return Random.SamplerSimple(
+        Random.SamplerType{T}(),
+        Random.SamplerBigFloat{Random.CloseOpen01{BigFloat}}(precision(T)),
+    )
+end
+
+function Random.Sampler(
+    RNG::Type{<:Random.AbstractRNG},
+    x::TOrRef,
+    n::Random.Repetition,
+) where {TOrRef<:Union{Arf,ArfRef,Arb,ArbRef,Acb,AcbRef}}
+    T = _nonreftype(TOrRef)
+    return Random.SamplerSimple(
+        Random.SamplerType{T}(),
+        Random.SamplerBigFloat{Random.CloseOpen01{BigFloat}}(precision(x)),
+    )
+end
```

Comment:
Should we have the ref types here? I didn't quite follow the discussion about why to remove then so I could be wrong.

Reply:
yes, I believe so:
* the first sampler is used when calling `rand(Arb, ...)`, i.e. **on type**: rand(::Type{T}); here julia-master typeasserts this call always to return object of type `T`, so we had conflict with `rand(ArbRef, ...)` returning honest `Arb`s.
* the second sampler is applicable when calling rand on value `rand(Arb(prec=64))`, but also `x = Acb(prec=64); rand(Arblib.realref(x),...)`

@rfourquet suggested this is not the most natural call, since value of `x` has nothing to do with the return value (it's there only for passing its precision), but it is what it is ;) @Joel-Dahne If you have a better idea how to handle this, let me know!

and yes, someone might follow interpret that since `Arb`s are intervals, `rand(Arb("1.1"))` should return an `Arf` in proximity of `1.1`. This would be against what `IntervalArithmetic` does though:
https://github.com/JuliaIntervals/IntervalArithmetic.jl/blob/cb987732830339a160c12c6d6af0ee2c0f0dbf21/src/rand.jl#L3

----
Another option would be to create `SamplerArb` cheap objects e.g. through `make` from `RandomExtensions` which then would carry the precision, but I think it's more natural to use value here.

Reply:
Oh I see, so it's still possible to use instances of `ArbRef` in the generation. This seems reasonable!

About this maybe not being the most natural call, since value of x has nothing to do with the return value (it's there only for passing its precision), I don't fully agree. It's in line with how `zero` and `one` works for example, though `zero` and `one` doesn't preserve the precision for `BigFloat`.